src/windows-emulator/syscalls/file.cpp

```diff
@@ -1067,21 +1069,74 @@ namespace syscalls
         return STATUS_NOT_SUPPORTED;
     }
 
-    NTSTATUS handle_NtCreateNamedPipeFile(
-        const syscall_context& c, const emulator_object<handle> /*file_handle*/, const ULONG /*desired_access*/,
-        const emulator_object<OBJECT_ATTRIBUTES<EmulatorTraits<Emu64>>> /*object_attributes*/,
-        const emulator_object<IO_STATUS_BLOCK<EmulatorTraits<Emu64>>> /*io_status_block*/, const ULONG /*share_access*/,
-        const ULONG /*create_disposition*/, const ULONG /*create_options*/, const ULONG /*named_pipe_type*/,
-        const ULONG /*read_mode*/, const ULONG /*completion_mode*/, const ULONG /*maximum_instances*/,
-        const ULONG /*inbound_quota*/, const ULONG /*outbound_quota*/,
-        const emulator_object<LARGE_INTEGER> /*default_timeout*/)
+    NTSTATUS handle_NtCreateNamedPipeFile(const syscall_context& c, emulator_object<handle> file_handle,
+                                          ULONG desired_access,
+                                          emulator_object<OBJECT_ATTRIBUTES<EmulatorTraits<Emu64>>> object_attributes,
+                                          emulator_object<IO_STATUS_BLOCK<EmulatorTraits<Emu64>>> io_status_block,
+                                          ULONG share_access, ULONG create_disposition, ULONG create_options,
+                                          ULONG named_pipe_type, ULONG read_mode, ULONG completion_mode,
+                                          ULONG maximum_instances, ULONG inbound_quota, ULONG outbound_quota,
+                                          emulator_object<LARGE_INTEGER> default_timeout)
     {
-        c.win_emu.log.error("Unimplemented syscall NtCreateNamedPipeFile!");
-        c.emu.stop();
+        (void)desired_access;
+        (void)share_access;
+        (void)create_disposition;
+        (void)create_options;
+        (void)object_attributes;
 
-        return STATUS_NOT_SUPPORTED;
+        std::u16string file_name;
+
+        // Get file name
+        object_attributes.access([&](const auto& attrs) {
+            emulator_object<UNICODE_STRING<EmulatorTraits<Emu64>>> unicode_string(
+                c.emu, static_cast<uint64_t>(attrs.ObjectName));
+
+            unicode_string.access([&](const auto& unicode) {
+                if (unicode.Length > 0 && unicode.Buffer != 0)
+                {
+                    auto buffer_addr = static_cast<uint64_t>(unicode.Buffer);
+                    std::vector<char16_t> buffer(unicode.Length / sizeof(char16_t));
+                    c.emu.read_memory(buffer_addr, buffer.data(), unicode.Length);
+                    file_name.assign(buffer.begin(), buffer.end());
+                }
+            });
+        });
+
+        //Build IO device
```

Comment:
don't use redundant comments please. the code already shows you are building the io device. it doesn't add any meaningful information